phoenicis-repository/src/main/java/org/phoenicis/repository/repositoryTypes/LocalRepository.java

```diff
@@ -101,8 +101,15 @@ public RepositoryDTO fetchInstallableApplications() {
 
                 if (categoryJson.exists()) {
                     final CategoryDTO.Builder categoryDTOBuilder = new CategoryDTO.Builder(
-                            unSerializeCategory(categoryJson)).withId(categoryDirectory.getName())
-                                    .withApplications(fetchApplications(categoryDirectory));
+                            unSerializeCategory(categoryJson)).withApplications(fetchApplications(categoryDirectory));
+
+                    if (StringUtils.isBlank(categoryDTOBuilder.getId())) {
```

Comment:
What would happen if `!StringUtils.isBlank(categoryDTOBuilder.getId())` is true and `categoryDTOBuilder.getId()` contains a space character?